src/pages/components/HistoryItem.qml

```diff
@@ -0,0 +1,106 @@
+import QtQuick 2.6
+import Sailfish.Silica 1.0
+import Sailfish.Browser 1.0
+
+ListItem {
+    id: root
+
+    property string search
+
+    function remove() {
+        view.model.remove(model.index)
+    }
+
+    function openLink() {
+        webView.load(model.url, model.title)
+        pageStack.pop()
+    }
+
+    width: view.width
+    menu: contextMenuComponent
+
+    ListView.onAdd: AddAnimation { target: root }
+
+    Column {
+        anchors {
+            left: parent.left
+            leftMargin: Theme.horizontalPageMargin
+            right: rightColumn.left
+            rightMargin: Theme.paddingSmall
+        }
+        HistoryLabel {
+            id: titleText
+            text: Theme.highlightText(model.title, search, Theme.highlightColor)
+        }
+
+        HistoryLabel {
+            text: Theme.highlightText(model.url, search, Theme.highlightColor)
+            opacity: Theme.opacityHigh
+        }
+    }
+    Column {
+        id: rightColumn
+        width: dateLabel.implicitWidth
+        anchors {
+            right: deleteButton.left
+            rightMargin: Theme.paddingSmall
+        }
+        HistoryLabel {
+            id: dateLabel
+            text: Format.formatDate(model.date, Formatter.DateMediumWithoutYear)
+            opacity: Theme.opacityHigh
+        }
+
+        HistoryLabel {
+            text: Format.formatDate(model.date, Formatter.TimeValueTwentyFourHours)
+            opacity: Theme.opacityHigh
+        }
+    }
+    IconButton {
+        id: deleteButton
+        icon.source: "image://theme/icon-m-clear"
+        anchors {
+            verticalCenter: parent.verticalCenter
+            right: parent.right
+            rightMargin: Theme.horizontalPageMargin
+        }
+        onClicked: remove()
+    }
+
+    ListView.onRemove: animateRemoval()
+    onClicked: openLink()
+
+    Component {
+        id: contextMenuComponent
+
+        ContextMenu {
+            MenuItem {
+                //% "Open in private tab"
+                text: qsTrId("sailfish_browser-me-open-private-tab")
+                onClicked: {
+                    webView.privateMode = true
+                    openLink()
```

Comment:
We do not have concept of private tab -- only private and non private mode. I think this deserves a bit more so that "private" flag would be passed to the newly created page (and created so regardless of active model) and the model into which it will be added would be handled separately.

Reply:
Discussed this with design and this clearly needs more thinking.

1) Transition from non-private to private whilst opening a tab
2) When transition happens people will wonder where did my tabs went
3) No new tabs created... always replacing previously opened page => based on the menu item text I would assume new tabs.